learning-another-language/js-loops.md

```diff
@@ -0,0 +1,417 @@
+  // do stuff
+}
+```
+
+The `for-loop` consists of three optional expressions, followed by a block of code:
+
+- **initialization** - This expression runs before the execution of the first loop. The initialization expression is usually used to create a counter.
+- **condition** - This expression is checked each time before the loop runs. If it evaluates to true, the code in the loop is executed. If it evaluates to false, the loop stops. If this expression is omitted, it automatically evaluates to true, and the code block is executed.
+- **finalExpression** - This expression is executed after each iteration of the loop. This is usually used to increment (or increase) a counter, but can also be used to decrement (or decrease) a counter.
+
+### Examples
+
+**1. Print integers 1 .. 5**
+```js
+for (let i = 1; i <= 5; i++) {
+	console.log(i)
+}
+
+// Output:
+// 1
+// 2
+// 3
+// 4
+// 5
+
+// intialization statement: let i = 1
+// condition statement: i <= 5
+// final expression: i++, equivalent to i = i + 1
+```
+
+**2. Use a break statement to exit the loop immediately**
+```js
+for (let i = 1; i <= 5; i++) {
+	if (i == 4) {
+		break;
+	}
+	console.log(i)
+}
+
+// Output:
+// 1
+// 2
+// 3
+```
+
+### Common Pitfall: Exceeding the Bounds of an Array
+```js
+const arr = [ "foo", "bar", "baz" ];
+
+for (let i = 0; i <= arr.length; i++) {
+	console.log(arr[i]);
+}
+
+// Output:
+// foo
+// bar
+// baz
+// undefined
+
+// In the above example, arr.length = 3, 
+// so accessing arr[3] leads to an undefined element
+```
+
+When iterating over an array, it is possible to accidentally exceed the bounds of the array. One thing to note about the indices of an array: 
+
+- The first element of an array is stored at index 0. This is true for JavaScript as well as most other programming languages. Therefore, the index of the last element of an array is equal to `arr.length - 1`.
+
+When constructing a `for-loop`, there are a couple ways to handle exceeding the bounds of an array. They both involve altering the conditional statement in the `for-loop` to either:
+
+- `i < arr.length` or
+- `i <= arr.length - 1`
+
+```js
+const arr = [ "foo", "bar", "baz" ];
+
+// Replace the <= with < in the condition statement
+for (let i = 0; i < arr.length; i++) {
+	console.log(arr[i]);
+}
+
+// Output:
+// foo
+// bar
+// baz
+
+// Replace arr.length with arr.length - 1
+for (let i = 0; i <= arr.length - 1; i++) {
+	console.log(arr[i]);
+}
+
+// Output:
+// foo
+// bar
+// baz
+```
+
+## 2.`for … of` loop
+```js
+for (variable of object) {
+	// do something
+}
+```
+
+The `for … of` loop can be used to iterate over arrays, sets, and maps. When using the `for … of` loop, there’s no danger of going out of bounds! The loop will go over each item in the iterable object.
+
+### Examples
+
+**1. Iterate over items in a list**
+```js
+const monsters = [
+	"Michael Myers", 
+	"Jason Voorhees", 
+	"Freddy Krueger"
+];
+
+for (let monster of monsters) {
```

Comment:
Consider expanding upon the `let  monster of monsters` syntax. This may look unfamiliar because Python doesn't require any keywords to define a new variable. May also be a good opportunity to reinforce previous material on `let` vs `var` and `const`. Why use `let` here?